[2024-06-30T08:35:29.950+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-30T08:35:29.975+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: northwindDWH.data_delivery manual__2024-06-30T08:35:26.435492+00:00 [queued]>
[2024-06-30T08:35:29.983+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: northwindDWH.data_delivery manual__2024-06-30T08:35:26.435492+00:00 [queued]>
[2024-06-30T08:35:29.983+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-06-30T08:35:29.997+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): data_delivery> on 2024-06-30 08:35:26.435492+00:00
[2024-06-30T08:35:30.007+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=123) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-30T08:35:30.009+0000] {standard_task_runner.py:63} INFO - Started process 125 to run task
[2024-06-30T08:35:30.009+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'northwindDWH', 'data_delivery', 'manual__2024-06-30T08:35:26.435492+00:00', '--job-id', '84', '--raw', '--subdir', 'DAGS_FOLDER/northwindDWH.py', '--cfg-path', '/tmp/tmpgoj860sl']
[2024-06-30T08:35:30.011+0000] {standard_task_runner.py:91} INFO - Job 84: Subtask data_delivery
[2024-06-30T08:35:30.024+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  SQL_ALCHEMY_CONN = conf.get("database", "SQL_ALCHEMY_CONN")

[2024-06-30T08:35:30.062+0000] {task_command.py:426} INFO - Running <TaskInstance: northwindDWH.data_delivery manual__2024-06-30T08:35:26.435492+00:00 [running]> on host 3abb3aef8f21
[2024-06-30T08:35:30.163+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Mahmoud' AIRFLOW_CTX_DAG_ID='northwindDWH' AIRFLOW_CTX_TASK_ID='data_delivery' AIRFLOW_CTX_EXECUTION_DATE='2024-06-30T08:35:26.435492+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-30T08:35:26.435492+00:00'
[2024-06-30T08:35:30.164+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-30T08:35:30.183+0000] {northwindDWH.py:49} INFO - Data delivery started
[2024-06-30T08:35:30.184+0000] {Load_data.py:85} INFO - loading dimEmployee to db ...
[2024-06-30T08:35:30.184+0000] {Transform_data.py:49} INFO - start Extracting dim employee
[2024-06-30T08:35:30.188+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.191+0000] {Transform_data.py:58} INFO - Extract dim employees done
[2024-06-30T08:35:30.192+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.196+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.199+0000] {Load_data.py:103} INFO - SCD type 2 on dimemployee started
[2024-06-30T08:35:30.206+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.206+0000] {Load_data.py:158} INFO - Loading dimproduct to db...
[2024-06-30T08:35:30.207+0000] {Transform_data.py:26} INFO - Start Extracting dim product
[2024-06-30T08:35:30.209+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.212+0000] {Transform_data.py:40} INFO - Extract dim products done
[2024-06-30T08:35:30.212+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.215+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.220+0000] {Load_data.py:170} INFO - SCD type 1 on dimproduct started
[2024-06-30T08:35:30.236+0000] {Tables_creations.py:27} INFO - Loading data into table dim_products...
[2024-06-30T08:35:30.239+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.246+0000] {Load_data.py:201} ERROR - Error while loading dimProducts: integer out of range

[2024-06-30T08:35:30.247+0000] {Load_data.py:12} INFO - loading dimCustomers to db ...
[2024-06-30T08:35:30.247+0000] {Transform_data.py:7} INFO - start Extracting dim customers
[2024-06-30T08:35:30.251+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.253+0000] {Transform_data.py:16} INFO - Extract dim customer done
[2024-06-30T08:35:30.254+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.258+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.261+0000] {Load_data.py:30} INFO - SCD type 2 on dimcustomer started
[2024-06-30T08:35:30.276+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.277+0000] {Load_data.py:205} INFO - Loading factsales to db ...
[2024-06-30T08:35:30.277+0000] {Transform_data.py:68} INFO - start Extracting fact sales
[2024-06-30T08:35:30.280+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.285+0000] {Transform_data.py:79} INFO - Extract fact sales done
[2024-06-30T08:35:30.285+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.289+0000] {Tables_creations.py:27} INFO - Loading data into table fact_sales...
[2024-06-30T08:35:30.292+0000] {db_utils.py:21} INFO - Connected to PostgreSQL database successfully
[2024-06-30T08:35:30.311+0000] {Tables_creations.py:41} INFO - Data loaded successfully into table fact_sales With 58 Row
[2024-06-30T08:35:30.311+0000] {db_utils.py:30} INFO - PostgreSQL connection is closed
[2024-06-30T08:35:30.312+0000] {Load_data.py:213} INFO - Fact_Sales loaded successfully
[2024-06-30T08:35:30.312+0000] {northwindDWH.py:54} INFO - Data delivery done
[2024-06-30T08:35:30.312+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-06-30T08:35:30.313+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-30T08:35:30.325+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=northwindDWH, task_id=data_delivery, run_id=manual__2024-06-30T08:35:26.435492+00:00, execution_date=20240630T083526, start_date=20240630T083529, end_date=20240630T083530
[2024-06-30T08:35:30.384+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-06-30T08:35:30.400+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-30T08:35:30.402+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
